### DISCUSSION TITLE
[Question] Is there an alternative to the global/static AnsiConsole for testing?

### DISCUSSION THREAD

**Discussion by wischi-chr** [spectreconsole/spectre.console#571] [Q&A] [answered]
I'm not sure if that's the right place to asks questions but I couldn't find anything about where to ask.

Is there a way to get an instance for `AnsiConsole` instead of using the static class? Or how would it be possible to run unit tests in parallel with a static global state? For example calling `AnsiConsole.ResetColors()` would reset the current style for all tests that run at the same time. Is there a way around that I'm missing?

**Comment by phil-scott-78** [comment, marked as answer]
Yup. Take a look at IAnsiConsole. Right now there isn't a standard mock or fake around it but you can look at the unit tests in the source and see how it's being used to for testing the library itself.

**Comment by wischi-chr** [comment]
Thank you. I will take a look at it.
